### ISSUE TITLE
Bug - Codes: One or multiple codes causes Sonic to freeze in place during & after environment monologues.

### ISSUE THREAD

**Issue by Optimus-Bull** [hedge-dev/HMMCodes#83]
As the title describes it.

One or multiple codes seem to make Sonic freeze in place during and after the environmental monologues throughout the game. Eggman sites being notable offenders of this. Therefore not being able to move at all again after the monologue ends.

This is regardless of whether or not the "Hold on Monologue" is enabled in the game settings.
Opening the skills menu, entering & exiting the training simulator is a viable workaround, though it doesn't solve the issue at the root.
Disabling all codes does fix the issue.

Haven't been able to nail down which code is causing this yet, as I have a lot of them being enabled.
I'm currently in the process of eliminating codes one by one to determine which one could be a potential troublemaker.

**Comment by DeaTh-G** [comment]
I'll mark the issue with a bug label, however it is not yet viable for us to fix this issue without further information so the issue will also get a blocked label in the time being.

**Comment by HyperBE32** [comment]
This sounds similar to an issue reported for v1.40 with `Retain Horizontal Velocity from Jump` (https://github.com/hedge-dev/HMMCodes/issues/23) - please confirm if this is the case here.

**Comment by Optimus-Bull** [comment]
I don't have that specific code enabled.

But you did gave me an idea on where to start, as I could test this further by first disabling the other "Retain Horizontal Velocity" codes.

I will update the bug report as soon as I find something.

**Comment by Optimus-Bull** [comment]
I looked at the thread RinMaru created. Rin only had Reduced homing delay, tighter drop dash & tighter jump rotation enabled and still had the issue.

So I started disabling those 3, one by one until I only had Reduced Homing Delay enabled.
Reduced Homing Delay has been consistent so far in triggering the issue where Sonic isn't allowed to move after the monologues.

**Comment by Optimus-Bull** [comment]
Wait, I'm still not completely sure Reduced Homing Delay is the culprit.

While it did remove the bug after I disabled it, Reduced Homing Delay also wouldn't cause the bug when being enabled by itself without any codes.

So it could be codes in tandem with each other that causes this.

**Comment by Optimus-Bull** [comment]
This is getting more and more difficult.
By each code I disable in HedgeModManager, the bigger the threshold becomes for triggering the bug.

I can confirm that none of the codes in the Physics section seems to cause the issue.
Air boost and stomp reset also does not cause the issue described.

**Comment by Optimus-Bull** [comment]
<img width="611" alt="HedgeModManager_Codes" src="https://github.com/hedge-dev/HMMCodes/assets/106356333/0fe618c8-8250-4d60-bc46-a085f6b5e5e2">

This is the current list of enabled codes where one or more is still causing the bug.
Any suggestions which code could be a troublemaker?

**Comment by Optimus-Bull** [comment]
I might have a code worth looking more into.
"Always Dodge in Stick Direction"

I tested back and forth enabling and disabling the code. Managed to trigger the bug with the code, but where completely unable to do so with it disabled.

I'm not fully convinced though. And I believe more than one code could trigger this bug. Especially gameplay or physics code the specifically alters Sonic's moveset.
